src/nvim/diff.c

```diff
@@ -3068,17 +3068,17 @@ static int parse_diff_ed(char_u *line, diffhunk_T *hunk)
   }
 
   if (difftype == 'a') {
-    hunk->lnum_orig = f1 + 1;
+    hunk->lnum_orig = (linenr_T)f1 + 1;
```

Comment:
f1 is `long` so why is this cast necessary? oh because `long` could be 64 wide :(

Does changing `f1` to int32 avoid the casts? Else we may need to assert...

Reply:
Yes, but need to introduce another cast.